x/sets/bloom_filter.go

```diff
@@ -0,0 +1,151 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package xsets
+
+import (
+	"math"
+
+	xmurmur3 "github.com/m3db/m3db/x/murmur3"
+)
+
+// BloomFilter is a bloom filter set membership.
+type BloomFilter struct {
+	m   uint64
+	k   uint64
+	set *BitSet
+}
+
+// NewBloomFilter creates a new bloom filter that can represent
+// m elements with k hashes.
+func NewBloomFilter(m uint, k uint) *BloomFilter {
+	if m < 1 {
```

Comment:
I wonder if it should return an error if these values are invalid...might be annoying to add the error value everywhere though

Reply:
Possibly, although am definitely trying to keep the generic datastructures as simple as possible to use so its more ergonomic.  I think in this case keeping both bitset and bloomfilter without an error is more ergonomic currently.  If there was some other args that could truly screw up things then at that point I'd lean towards an error here.  

Overall a good comment but I think just for ergonomics we should probably avoid returning an error for this API.